operator/pkg/reconciliation/decommission_node.go

```diff
@@ -0,0 +1,327 @@
+					return result.Error(err)
+				}
+			}
+
+			rc.ReqLogger.Info(
+				"Need to update the rack's node count",
+				"Rack", rackInfo.RackName,
+				"maxReplicas", maxReplicas,
+				"desiredSize", desiredNodeCount,
+			)
+
+			rc.Recorder.Eventf(rc.Datacenter, corev1.EventTypeNormal, events.ScalingDownRack,
+				"Scaling down rack %s", rackInfo.RackName)
+
+			if err := setOperatorProgressStatus(rc, api.ProgressUpdating); err != nil {
+				return result.Error(err)
+			}
+
+			err := rc.DecommissionNodeOnRack(rackInfo.RackName, epData, lastPodSuffix)
+			if err != nil {
+				return result.Error(err)
+			}
+
+			return result.RequeueSoon(10)
+		}
+	}
+
+	return result.Continue()
+}
+
+func (rc *ReconciliationContext) DecommissionNodeOnRack(rackName string, epData httphelper.CassMetadataEndpoints, lastPodSuffix string) error {
+	for _, pod := range rc.dcPods {
+		podRack := pod.Labels[api.RackLabel]
+		if podRack == rackName && strings.HasSuffix(pod.Name, lastPodSuffix) {
+			mgmtApiUp := isMgmtApiRunning(pod)
+			if !mgmtApiUp {
+				return fmt.Errorf("Management API is not up on node that we are trying to decommission")
+			}
+
+			if err := rc.EnsurePodsCanAbsorbDecommData(pod, epData); err != nil {
+				dcPatch := client.MergeFrom(rc.Datacenter.DeepCopy())
+				updated := rc.setCondition(api.NewDatacenterCondition(api.DatacenterScaleDownFailed,
+					corev1.ConditionTrue))
+
+				if updated {
+					patchErr := rc.Client.Status().Patch(rc.Ctx, rc.Datacenter, dcPatch)
+					if patchErr != nil {
+						rc.ReqLogger.Error(patchErr, "error patching datacenter status for scale down failed")
+						return patchErr
+					}
+				}
+				return err
+			}
+
+			if err := rc.NodeMgmtClient.CallDecommissionNodeEndpoint(pod); err != nil {
+				// TODO this returns a 500 when it works
+				// We are waiting for a new version of mgmt api with a fix for this
+				// return false, err
+			}
+
+			rc.ReqLogger.Info("Marking node as decommissioning")
+			patch := client.MergeFrom(pod.DeepCopy())
+			pod.Labels[api.CassNodeState] = stateDecommissioning
+			if err := rc.Client.Patch(rc.Ctx, pod, patch); err != nil {
+				return err
+			}
+
+			rc.Recorder.Eventf(rc.Datacenter, corev1.EventTypeNormal, events.LabeledPodAsDecommissioning,
+				"Labeled node as decommissioning %s", pod.Name)
+
+			return nil
+		}
+	}
+
+	// this shouldn't happen
+	return fmt.Errorf("Could not find pod to decommission on rack %s", rackName)
+}
+
+// Wait for decommissioning nodes to finish before continuing to reconcile
+func (rc *ReconciliationContext) CheckDecommissioningNodes(epData httphelper.CassMetadataEndpoints) result.ReconcileResult {
+	if rc.Datacenter.GetConditionStatus(api.DatacenterScalingDown) == corev1.ConditionFalse {
+		return result.Continue()
+	}
+
+	for _, pod := range rc.dcPods {
+		if pod.Labels[api.CassNodeState] == stateDecommissioning {
+			if !IsDoneDecommissioning(pod, epData) {
```

Comment:
Don't forget to clear the pod from `CassandraDatacenter.Status.NodeStatuses`

Reply:
Thanks for the reminder :)